pilot/pkg/model/sidecar.go

```diff
@@ -566,29 +579,55 @@ func (sc *SidecarScope) AddConfigDependencies(dependencies ...ConfigHash) {
 }
 
 // DestinationRule returns a destinationrule for a svc.
-func (sc *SidecarScope) DestinationRule(direction TrafficDirection, proxy *Proxy, svc host.Name) *ConsolidatedDestRule {
-	destinationRules := sc.destinationRules[svc]
-	var catchAllDr *ConsolidatedDestRule
+func (sc *SidecarScope) DestinationRule(direction TrafficDirection, proxy *Proxy, svc *Service) *ConsolidatedDestRule {
+	if svc == nil {
+		return nil
+	}
+
+	// select destinationrule with priority order:
+	// 1. same namespace as the proxy, and with workload selector
+	// 2. same namespace as the proxy
+	var proxyNamespaceDrs []*ConsolidatedDestRule
+	destinationRules := sc.destinationRules[svc.Hostname][sc.Namespace]
 	for _, destRule := range destinationRules {
 		destinationRule := destRule.rule.Spec.(*networking.DestinationRule)
 		if destinationRule.GetWorkloadSelector() == nil {
-			catchAllDr = destRule
+			proxyNamespaceDrs = append(proxyNamespaceDrs, destRule)
 		}
-		// filter DestinationRule based on workloadSelector for outbound configs.
-		// WorkloadSelector configuration is honored only for outbound configuration, because
-		// for inbound configuration, the settings at sidecar would be more explicit and the preferred way forward.
-		if sc.Namespace == destRule.rule.Namespace &&
-			destinationRule.GetWorkloadSelector() != nil && direction == TrafficDirectionOutbound {
+		if destinationRule.GetWorkloadSelector() != nil && direction == TrafficDirectionOutbound {
 			workloadSelector := labels.Instance(destinationRule.GetWorkloadSelector().GetMatchLabels())
 			// return destination rule if workload selector matches
 			if workloadSelector.SubsetOf(proxy.Labels) {
+				// 1
```

Comment:
can you please add comment for 1 here - it will be easy to follow if it is here?

Reply:
yes, i will do